runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/impl/StandardRetryStrategy.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.retries.impl
+
+import aws.smithy.kotlin.runtime.retries.*
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.withTimeout
+
+/**
+ * Implements a retry strategy utilizing backoff delayer and a token bucket for rate limiting and circuit breaking. Note
+ * that the backoff delayer and token bucket work independently of each other. Either can delay retries (and the token
+ * bucket can delay the initial try). The delayer is called first so that the token bucket can refill as appropriate.
+ * @param options The options that control the functionality of this strategy.
+ * @param tokenBucket The token bucket instance. Utilizing an existing token bucket will share call capacity between
+ * those scopes.
+ * @param backoffDelayer A delayer that can back off after the initial try to spread out the retries.
+ */
+class StandardRetryStrategy(
```

Comment:
suggestion

Rather than `StandardRetryStrategy` why not just name it `TokenBucketRetryStrategy`?

Reply:
I liked the name "standard" because it implied _recommended_. Also, the retry specification calls it "standard". I don't mind renaming it to something clearer but I don't like `TokenBucketRetryStrategy` given that there's a `StandardRetryTokenBucket`...any other ideas?

Reply:
Yeah I kind of started to understand the naming came from the SEP as I was reading. 

That being said it's only "standard" because of the values chosen for the configuration and the implementation of `TokenBucket` given as a parameter. Other than that it's an implementation of a retry strategy using a token bucket. I'm not sure it matters all that much but my preference I guess is to rename this to `TokenBucketRetryStrategy`. The `StandardRetryTokenBucket` makes more sense because it is what actually implements the "standard" logic.

Not a huge deal just my 2 cents